### ISSUE TITLE
[BUG] 404 Not Found for manually deleted resources

### ISSUE THREAD

**Issue by sprinthive-kieran** [mezmo/terraform-provider-mezmo#109]
Hi Team,

I have encountered what I suspect is a bug:

When a resource is created via Terraform and then manually deleted using the Mezmo portal, any subsequent plans or applies fail with the error `404 Not Found`

An example with `mezmo_pipeline`:
```
│ Error: Error Reading Pipeline
│ 
│   with module.mezmo_pipeline.mezmo_pipeline.auth0,
│   on ../../../../../modules/mezmo/pipelines/auth0/main.tf line 12, in resource "mezmo_pipeline" "auth0":
│   12: resource "mezmo_pipeline" "auth0" {
│ 
│ Could not read pipeline with id  xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxx: 404 Not Found: 
```

An example with `http_source`:
```
│ Error: Error Reading Source
│ 
│   with module.mezmo_pipeline.mezmo_http_source.auth0_stream,
│   on ../../../../../modules/mezmo/pipelines/auth0/main.tf line 16, in resource "mezmo_http_source" "auth0_stream":
│   16: resource "mezmo_http_source" "auth0_stream" {
│ 
│ Could not read source with id "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxx" and pipeline_id "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxx": 404 Not Found: 
```

I would expect the provider to handle these errors, and attempt to recreate any deleted resources.

This current behaviour requires using `terraform state rm` to remove all deleted resources in order to return to a sane state, which is less than ideal, especially when dealing with a complex pipeline with many components that have been deleted.

I apologise if this is the wrong forum to report bugs, please let me know if there is a more suitable one I should rather post on.

Thanks!

**Comment by dhable** [comment]
Thank you for reporting the bug. It appears that this use case was overlooked when we were building out the provider. As you mentioned, removing the missing resources from the state and reapplying the change should get you back and running.

I've created an internal ticket, LOG-19148, to get this fix scheduled and assigned to a developer.

**Comment by dhable** [comment]
This was addressed in the [v1.12.0 release](https://github.com/mezmo/terraform-provider-mezmo/releases/tag/v1.12.0), that was just published 3 hours ago.
